File: linked lists/creating_a_list(inserting a node at the end)/creating_a_list.c
#include"stdio.h"
#include"stdlib.h"

void print_list(void);
void insert_node(void);

typedef struct
{
    int data;
    struct Node *next ;
}Node;

Node * create_node(void);

Node *start = NULL;

void main()
{
    int n;
    printf("Do you want to make a linked list?\n1. YES\n2. NO\n");
    scanf("%d",&n);
    while(1)
    {
        switch(n)
        {

            case 1:
                {

                insert_node();
                printf("Do you want to continue?\n1. YES\n2. NO\n");
                scanf("%d",&n);
                break;
                }

            case 2:{
                    print_list();
                    exit(0);
            }

            default:printf("Invalid selection\n");
            }
        }

}


Node * create_node()
{
    Node *n;
    n = (Node *)malloc(sizeof(Node));
    return(n);
}


void insert_node()
{
    Node *temp, *t;
    temp = create_node();
    temp->next = NULL;
    printf("Enter a number\n");
    scanf("%d",&temp->data);
    if(start == NULL)
    {
        start = temp;
    }
    else
    {
        t = start;
        while(t->next != NULL)
        {
            t = t->next;
        }
        t->next = temp;
    }
}


void print_list()
{
    Node *p;
    printf("Your final linked list is \n");
    p = start;
    while(p->next != NULL)
    {
        printf("%d\n",p->data);
        p = p->next;
    }
    printf("%d\n",p->data);

}

File: linked lists/creating_a_list(inserting a node at the end)/creating_a_list.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Tue Aug 27 00:34:13 2019

@author: sandeep
"""

class Node():
    def __init__(self):
        self.data = None
        self.next = None

class LinkedList():
    def __init__(self):
        self.head = None
        
    def insert(self):
        num = int(input("Enter a number: "))
        node = Node()
        node.data = num
        
        if self.head == None:
            self.head = node
            return
        last = self.head
        
        while last.next:
            last = last.next
        
        last.next = node
        
    def print_list(self):
        
        current = self.head
        
        
        print("\nYour final list is:")
        
        if self.head == None:
            print("List is empty\n")
            return
        
        while current:
            print(current.data)
            current = current.next
        
        
        
            



my_list = LinkedList()

while True:
    print("\nWhat do you want to do?\n1. Insert at end\n2. Print list\n3. End")
    choice = int(input("Enter your choice: "))
    
    if choice == 1:
        my_list.insert()
        
    elif choice == 2:
        my_list.print_list()
        
    elif choice == 3:
        break
    else:
        print("Invalid Choice\n")
    

















